core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -365,6 +365,29 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
       case ConfigType.Ip => changeIpConfig(entityName, configs)
       case _ => throw new IllegalArgumentException(s"$entityType is not a known entityType. Should be one of ${ConfigType.all}")
     }
+
```

Comment:
It seems that the switch/match block above ends up calling `changeEntityConfig` regardless of the entity type. As a final step, `changeEntityConfig` creates a ZK notification. Would it be better to perform the cleanup before the ZK notification is triggered?

Reply:
Agree with that. I have moved the clean logic to method changeEntityConfig before zk notification is triggered.